httpclient-processor/src/main/java/org/springframework/cloud/stream/module/httpclient/HttpClientProcessorProperties.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.http.HttpMethod;
+
+/**
+ * Configuration properties for the Http Client Processor module.
+ *
+ * @author Waldemar Hummer
+ * @author Mark Fisher
+ */
+@ConfigurationProperties
+public class HttpClientProcessorProperties {
+
+	private static final HttpMethod DEFAULT_HTTP_METHOD = HttpMethod.GET;
+
+	private static final Class<?> DEFAULT_RESPONSE_TYPE = String.class;
+
+	public static final Expression DEFAULT_BODY_EXPRESSION = new SpelExpressionParser().parseExpression("payload");
+
+	/**
+	 * The URL to issue an http request to, as a static value.
+	 */
+	private String url;
+
+	/**
+	 * A SpEL expression against incoming message to determine the URL to use.
+	 */
+	private Expression urlExpression;
+
+	/**
+	 * The kind of http method to use.
+	 */
+	private HttpMethod httpMethod = DEFAULT_HTTP_METHOD;
+
+	/**
+	 * The (static) body of the request to use.
+	 */
+	private Object body;
+
+	/**
+	 * A SpEL expression against incoming message to derive the request body to use.
+	 */
+	private Expression bodyExpression = DEFAULT_BODY_EXPRESSION;
+
+	/**
+	 * A SpEL expression used to derive the http headers map to use.
+	 */
+	private Expression headersExpression;
+
+	/**
+	 * The type used to interpret the response.
+	 */
+	private Class<?> expectedResponseType = DEFAULT_RESPONSE_TYPE;
+
+	/**
+	 * A SpEL expression used to compute the final result, applied against the whole http response.
+	 */
+	private Expression resultExpression = new SpelExpressionParser().parseExpression("body");
+
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+	public Expression getUrlExpression() {
+		return urlExpression != null ? urlExpression
+				: new LiteralExpression(this.url);
+	}
+
+	public void setUrlExpression(Expression urlExpression) {
+		this.urlExpression = urlExpression;
+	}
+
+	@NotNull
+	public HttpMethod getHttpMethod() {
+		return httpMethod;
+	}
+
+	public void setHttpMethod(HttpMethod httpMethod) {
+		this.httpMethod = httpMethod;
+	}
+
+	public Object getBody() {
+		return body;
+	}
+
+	public void setBody(Object body) {
+		if (bodyExpression != null &&
+				bodyExpression.getExpressionString().equals(DEFAULT_BODY_EXPRESSION.getExpressionString())) {
+			this.bodyExpression = null;
+		}
+		this.body = body;
+	}
+
+	public Expression getBodyExpression() {
+		return bodyExpression;
+	}
+
+	public void setBodyExpression(Expression bodyExpression) {
+		this.bodyExpression = bodyExpression;
+	}
+
+	public Expression getHeadersExpression() {
+		return headersExpression;
+	}
+
+	public void setHeadersExpression(Expression headersExpression) {
+		this.headersExpression = headersExpression;
+	}
+
+	@NotNull
+	public Class<?> getExpectedResponseType() {
+		return expectedResponseType;
+	}
+
+	public void setExpectedResponseType(Class<?> expectedResponseType) {
+		this.expectedResponseType = expectedResponseType;
+	}
+
+	@NotNull
+	public Expression getResultExpression() {
```

Comment:
Had the exact same train of thoughts. But "response" is used in the _http response_ context, so I changed to `result`